hypothesis-python/docs/reference/api.rst

```diff
@@ -0,0 +1,1038 @@
+
+.. autoclass:: hypothesis.Phase
+   :members:
+
+The phases argument accepts a collection with any subset of these. e.g.
+``settings(phases=[Phase.generate, Phase.shrink])`` will generate new examples
+and shrink them, but will not run explicit examples or reuse previous failures,
+while ``settings(phases=[Phase.explicit])`` will only run the explicit
+examples.
+
+.. _verbose-output:
+
+Seeing intermediate result
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To see what's going on while Hypothesis runs your tests, you can turn
+up the verbosity setting.
+
+.. code-block:: pycon
+
+    >>> from hypothesis import find, settings, Verbosity
+    >>> from hypothesis.strategies import lists, integers
+    >>> @given(lists(integers()))
+    ... @settings(verbosity=Verbosity.verbose)
+    ... def f(x):
+    ...     assert not any(x)
+    ... f()
+    Trying example: []
+    Falsifying example: [-1198601713, -67, 116, -29578]
+    Shrunk example to [-1198601713]
+    Shrunk example to [-1198601600]
+    Shrunk example to [-1191228800]
+    Shrunk example to [-8421504]
+    Shrunk example to [-32896]
+    Shrunk example to [-128]
+    Shrunk example to [64]
+    Shrunk example to [32]
+    Shrunk example to [16]
+    Shrunk example to [8]
+    Shrunk example to [4]
+    Shrunk example to [3]
+    Shrunk example to [2]
+    Shrunk example to [1]
+    [1]
+
+The four levels are quiet, normal, verbose and debug. normal is the default,
+while in quiet mode Hypothesis will not print anything out, not even the final
+falsifying example. debug is basically verbose but a bit more so. You probably
+don't want it.
+
+If you are using :pypi:`pytest`, you may also need to
+:doc:`disable output capturing for passing tests <pytest:how-to/capture-stdout-stderr>`.
+
+Building settings objects
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Settings can be created by calling :class:`~hypothesis.settings` with any of the available settings
+values. Any absent ones will be set to defaults:
+
+.. code-block:: pycon
+
+    >>> from hypothesis import settings
+    >>> settings().max_examples
+    100
+    >>> settings(max_examples=10).max_examples
+    10
+
+You can also pass a 'parent' settings object as the first argument,
+and any settings you do not specify as keyword arguments will be
+copied from the parent settings:
+
+.. code-block:: pycon
+
+    >>> parent = settings(max_examples=10)
+    >>> child = settings(parent, deadline=None)
+    >>> parent.max_examples == child.max_examples == 10
+    True
+    >>> parent.deadline
+    200
+    >>> child.deadline is None
+    True
+
+Default settings
+~~~~~~~~~~~~~~~~
+
+At any given point in your program there is a current default settings,
+available as ``settings.default``. As well as being a settings object in its own
+right, all newly created settings objects which are not explicitly based off
+another settings are based off the default, so will inherit any values that are
+not explicitly set from it.
```

Comment:
iirc this is actually wrong: `settings.default` is the _default settings_, which are immutable, and you can access the _current settings_ by constructing a new object with `settings()`.  [whoops 😓]

Reply:
they're actually both equivalent! settings.default gets you the current settings as well. We should recommend one and only one correct way to do this though. I'd vote for `settings()`, because `settings.default` implies, well, the default settings, not the current settings.